In [32]:
import pandas as pd
import numpy as np

In [2]:
dfr= pd.read_pickle('D:/Capstone/dataset/database/messages_combined_reduced.pkl')
df= pd.read_pickle('D:/Capstone/dataset/database/messages_combined.pkl')
alias= pd.read_csv('D:/Capstone/dataset/database/N4J/alias.csv')

In [3]:
print(len(dfr))
print(len(df))

144785
473749


In [4]:
user_dict=dict(zip(alias['uuid'].unique().tolist(), alias['user'].unique()))
alias.drop(columns = ['Unnamed: 0'],inplace=True)
a_list=alias['eid'].unique().tolist()
print(len(a_list))

212


In [5]:
dfr_a=pd.DataFrame()
for a in a_list:
    result = dfr[(dfr['eid']==a)]
    dfr_a=dfr_a.append(result)

In [6]:
dfr_a.shape[0]

73596

In [7]:
dfr_a.to_csv('D:/Capstone/dataset/database/N4J/internal.csv')

dfr does not have 'unclassified' topics, we would want to include these only for intra-network analysis.
Lets extract this from df where it includes all topics, including 'unclassified' ones.
Let us also include this into the reduced dataframe. 

In [8]:
dfr_rec=dfr[dfr.type=='received']
df_rec =df[df.type=='received']

In [9]:
dfr_rec_a= pd.DataFrame()
df_rec_a= pd.DataFrame()

In [10]:
for a in a_list:
    result = dfr_rec[(dfr_rec['eid']==a)]
    dfr_rec_a=dfr_rec_a.append(result)

for a in a_list:
    result = df[(df['eid']==a)]
    df_rec_a=df_rec_a.append(result)

In [11]:
print(len(dfr_rec_a))
print(len(df_rec_a))
print(len(dfr_rec_a.eid.unique()))
print(len(df_rec_a.eid.unique()))

5703
229826
177
212


In [12]:
print(len(dfr_rec_a.eid.unique()))
print(len(dfr_rec_a.uuid.unique()))

177
142


In [13]:
print(len(df_rec_a.eid.unique()))
print(len(df_rec_a.uuid.unique()))

212
150


In [14]:
missing_a=[i for i in df_rec_a.eid.unique().tolist() if i not in dfr_rec_a.eid.unique().tolist()]

In [15]:
missing_df=pd.DataFrame()
for m in missing_a:
    result = df_rec_a[(df_rec_a['eid']==m)&(df_rec_a['topic']=='unclassified')]
    missing_df=missing_df.append(result)

In [16]:
print(len(missing_df))
print(len(missing_df.uuid.unique()))
print(len(missing_df.eid.unique()))
print(len(missing_a))

10276
63
35
35


In [17]:
dfr_rec_a = pd.concat([dfr_rec_a,missing_df],axis=0)
print(len(dfr_rec_a.eid.unique()))
print(len(dfr_rec_a.uuid.unique()))

212
144


In [18]:
missing_u=[i for i in df_rec_a.uuid.unique().tolist() if i not in dfr_rec_a.uuid.unique().tolist()]

In [19]:
missing_df=pd.DataFrame()
for m in missing_u:
    result = df_rec_a[(df_rec_a['uuid']==m)&(df_rec_a['topic']=='unclassified')]
    missing_df=missing_df.append(result)

In [20]:
dfr_rec_a = pd.concat([dfr_rec_a,missing_df],axis=0)
print(len(dfr_rec_a.eid.unique()))
print(len(dfr_rec_a.uuid.unique()))

212
150


In [21]:
len(dfr_rec_a)

18784

In [22]:
dfr_rec_a=dfr_rec_a.merge(alias, on='eid')

In [23]:
dfr_rec_a.drop(columns = 'from_clean',inplace=True)
dfr_rec_a.columns = ['mid', 'address', 'user_rec', 'type', 'subject', 'body', 'date', 'bcc',
       'uuid_rec', 'eid_rec', 'topic', 'topic_h','user_sent','uuid_sent']

In [24]:
dfr_rec_a

mid                  address     user_rec      type  \
0      165536  phillip.allen@enron.com   dasovich-j  received   
1      165536  phillip.allen@enron.com   dasovich-j  received   
2      165526  phillip.allen@enron.com      ermis-f  received   
3      165526  phillip.allen@enron.com      ermis-f  received   
4      165526  phillip.allen@enron.com      ermis-f  received   
5      165536  phillip.allen@enron.com       kean-s  received   
6      165536  phillip.allen@enron.com       kean-s  received   
7      165536  phillip.allen@enron.com       kean-s  received   
8      165536  phillip.allen@enron.com       kean-s  received   
9      165536  phillip.allen@enron.com       kean-s  received   
10     165536  phillip.allen@enron.com       kean-s  received   
11     165536  phillip.allen@enron.com       kean-s  received   
12     165536  phillip.allen@enron.com    sanders-r  received   
13     165536  phillip.allen@enron.com    sanders-r  received   
14     165536  phillip.allen@enron.com    sanders-r  received   
15     165392  phillip.allen@enron.com      smith-m  received   
16     165392  phillip.allen@enron.com      smith-m  received   
17     165724  phillip.allen@enron.com     taylor-m  received   
18     165724  phillip.allen@enron.com     taylor-m  received   
19     102549       k..allen@enron.com     martin-t  received   
20      86289  jeff.dasovich@enron.com       kean-s  received   
21      86685  jeff.dasovich@enron.com       kean-s  received   
22      88585  jeff.dasovich@enron.com       kean-s  received   
23      88565  jeff.dasovich@enron.com       kean-s  received   
24      88505  jeff.dasovich@enron.com       kean-s  received   
25      88411  jeff.dasovich@enron.com       kean-s  received   
26      88382  jeff.dasovich@enron.com       kean-s  received   
27      88360  jeff.dasovich@enron.com       kean-s  received   
28      88291  jeff.dasovich@enron.com       kean-s  received   
29      88164  jeff.dasovich@enron.com       kean-s  received   
...       ...                      ...          ...       ...   
18846  133814    mark.fisher@enron.com    fischer-m      sent   
18847  133815    mark.fisher@enron.com    fischer-m      sent   
18848  133816    mark.fisher@enron.com    fischer-m      sent   
18849  133807    mark.fisher@enron.com    fischer-m      sent   
18850   61275    eric.linder@enron.com  salisbury-h  received   
18851   61274    eric.linder@enron.com  salisbury-h  received   
18852   61276    eric.linder@enron.com  williams-w3  received   
18853   61278    eric.linder@enron.com  williams-w3  received   
18854   61277    eric.linder@enron.com  williams-w3  received   
18855   61273    eric.linder@enron.com     linder-e      sent   
18856   61272    eric.linder@enron.com     linder-e      sent   
18857   61271    eric.linder@enron.com     linder-e      sent   
18858   61270    eric.linder@enron.com     linder-e      sent   
18859   61269    eric.linder@enron.com     linder-e      sent   
18860   61268    eric.linder@enron.com     linder-e      sent   
18861   61268    eric.linder@enron.com     linder-e      sent   
18862   61273    eric.linder@enron.com     linder-e      sent   
18863   61268    eric.linder@enron.com     linder-e      sent   
18864   61270    eric.linder@enron.com     linder-e      sent   
18865   61271    eric.linder@enron.com     linder-e      sent   
18866   61269    eric.linder@enron.com     linder-e      sent   
18867   61272    eric.linder@enron.com     linder-e      sent   
18868   61273    eric.linder@enron.com     linder-e      sent   
18869  198403  steven.merris@enron.com  salisbury-h  received   
18870  198402  steven.merris@enron.com    merriss-s      sent   
18871  198401  steven.merris@enron.com    merriss-s      sent   
18872  198400  steven.merris@enron.com    merriss-s      sent   
18873  198400  steven.merris@enron.com    merriss-s      sent   
18874  198401  steven.merris@enron.com    merriss-s      sent   
18875  198402  steven.merris@enron.com    mer

In [25]:
print(len(dfr_rec_a.eid_rec.unique()))
print(len(dfr_rec_a.uuid_rec.unique()))
print(len(dfr_rec_a.uuid_sent.unique()))

212
150
149


In [26]:
dfr_rec_a.to_csv('D:/Capstone/dataset/database/N4J/internal_received_sent.csv')

In [27]:
dfr_rec_a2= pd.DataFrame()
for a in a_list:
    result = dfr[(dfr['eid']==a)]
    dfr_rec_a2=dfr_rec_a2.append(result)

So far we have created csvs for uploading to n4j using the following parameters:

1) from or sent by people within the 150 list - internal.csv

2) from within the 150 list, excludes includes 'unclassfied' emails, exclude 3rd parties - internal_received_sent.csv

In [28]:
dfr_rec_a2=dfr_rec_a2[dfr_rec_a2.type=='received']

In [29]:
dfr_rec_a2
dfr_rec_a2.to_csv('D:/Capstone/dataset/database/N4J/internal_received2.csv')

In [30]:
dfr_rec_a3= pd.DataFrame()
for a in a_list:
    result = df[(df['eid']==a)&(df['type']=='received')]
    dfr_rec_a3=dfr_rec_a3.append(result)

In [31]:
dfr_rec_a3
dfr_rec_a2.to_csv('D:/Capstone/dataset/database/N4J/internal_received3.csv')